### PR TITLE
Adb 006 crud agents

### PR DESCRIPTION
**Pull request by MatthewCrossley**
This PR adds CRUD operations for agents.

* create (`/agent/`)
  * Pass the `callSign`, `firstName`, `lastName`, `dateofBirth` and `agentRank` via POST request
* read (`/agent/:callSign`)
  * Will return the `agentId`, `callSign` and `agentRank`
* update (`/agent/update`):
  * You can update the `callSign`
  * Pass this as JSON in the put request body
* delete (`/agent/delete`)
  * Specify an agent to delete via their `callSign`
  * Only admins will be allowed to delete other users

I also fixed an annoying issue in frontend where there was a `Components` and `components/navbar` folder. I moved everything into `Components`.

### CODE REVIEW

File: agent-discoveries-backend/src/routes/agent.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      await db('agents')
+        .where({ agentId: req.session.user?.agentId })
+        .update({ callSign })
+      res.status(200).send()
+    } catch (e) {
+      res.status(400).send()
+    }
+  })
+
+  router.put('/delete', auth.isLoggedIn, async (req, res) => {
```
Comment (AndrewYHuang) [inline agent-discoveries-backend/src/routes/agent.ts]: Ooh actually - we could do 
```
router.delete('/:callsign', ...
```
or
```
router.delete('/:agentId', ...
```
which would be a bit more of a RESTful style than having a /delete endpoint :)

**Review by AndrewYHuang — COMMENTED** [review]
Looks good, just a small comment

File: agent-discoveries-backend/src/routes/agent.ts (hunk 1 above)
Comment (MatthewCrossley) [reply agent-discoveries-backend/src/routes/agent.ts]: Certainly can do. I mostly copied this from the users CRUD PR so I can go back and update that one afterwards

Comment (AndrewYHuang) [reply agent-discoveries-backend/src/routes/agent.ts]: Ahh yes please! That would be great 😄

Comment (MatthewCrossley) [reply agent-discoveries-backend/src/routes/agent.ts]: Done

**Pull request merged**